sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -276,6 +317,52 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
     });
   }
 
+  /**
+   * Adjust the study breaks when there is a change in study end date
+   * during scholastic standing.
+   * @param studyBreaks current offering study break.
+   * @param newStudyEndDate newly changed/updated study end date.
+   * @returns adjusted study breaks.
+   */
+  adjustStudyBreaks(
+    studyBreaks: StudyBreak[],
+    newStudyEndDate: string,
+  ): StudyBreak[] {
+    return studyBreaks
+      ?.map((studyBreak) => {
+        if (isBeforeDate(newStudyEndDate, studyBreak.breakStartDate)) {
+          // Ignore the study break.
+          return;
+        }
+        if (
+          isBetweenPeriod(newStudyEndDate, {
+            startDate: studyBreak.breakStartDate,
+            endDate: studyBreak.breakEndDate,
+          })
+        ) {
+          // Adjust the study break.
+          const breakDays = dateDifference(
+            newStudyEndDate,
+            studyBreak.breakStartDate,
+          );
+          const eligibleBreakDays =
```

Comment:
i guess you can refactor this to 
const eligibleBreakDays = Math.min(breakDays, OFFERING_STUDY_BREAK_MAX_DAYS);